rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -130,17 +140,9 @@ pub struct StreamProcessor<TPayload: Clone> {
 impl<TPayload: Clone + 'static> StreamProcessor<TPayload> {
     pub fn new(
         consumer: Box<dyn Consumer<TPayload, Callbacks<TPayload>>>,
-        processing_factory: Box<dyn ProcessingStrategyFactory<TPayload>>,
+        consumer_state: Arc<Mutex<ConsumerState<TPayload>>>,
         dlq_policy: Option<DlqPolicy<TPayload>>,
     ) -> Self {
-        let consumer_state = Arc::new(Mutex::new(ConsumerState {
```

Comment:
what is the reason for making the ConsumerState public API? this seems like a fairly complicated API for end users (even if the Arc is hidden inside)

Reply:
I agree that it shouldn't be public at all. In #5124 I add a function for creating Kafka `StreamProcessor`s that completely hides this type. For `LocalConsumer` I assume it doesn't really matter because that's a testing thing, right?

Reply:
On a related note, everything thats purely for testing should probably be in a `testing` or similarly named submodule thats feature gated.

Reply:
I would stick with the old `StreamProcessor::new` -- this is a breaking change and I don't think the shortcut for kafka makes it better. In tests the API matters just as much, arguably we have more tests in the testsuite than consumers in prod.